fish-rust/src/trace.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{
+    common::{escape_string, EscapeFlags},
+    ffi::{self, parser_t, wcharz_t},
+    wchar::{self, wstr, L},
+    wchar_ffi::WCharToFFI,
+};
+use std::sync::atomic;
+
+#[cxx::bridge]
+mod ffi2 {
+    extern "C++" {
+        include!("wutil.h");
+        include!("parser.h");
+        type wcharz_t = super::wcharz_t;
+        type parser_t = super::parser_t;
+    }
+
+    extern "Rust" {
+        fn rust_trace_set_enabled(do_enable: bool);
+        fn rust_trace_enabled(parser: &parser_t) -> bool;
+        fn rust_trace_argv(parser: &parser_t, command: wcharz_t, args: &Vec<wcharz_t>);
+        fn rust_trace_if_enabled(parser: &parser_t, command: wcharz_t);
```

Comment:
If you've removed the existing/old `trace_xxx()` functions (and I see that you have by deleting `trace.cpp` and `trace.h`) there's no reason to name these `rust_trace_xxx()`. 

In this file, use our existing convention of suffixing the ffi functions with `_ffi` instead of prefixing them with `rust_` then in the `extern "Rust" { ... }` section prefix each `trace_xxx_ffi()` declaration with `[cxx_name = "trace_xxx"]`.

We'll know that any fns with `_ffi` in the name should be deleted when we're done with the transition, and by renaming the type with `cxx_name` the old name will still be used in the cpp codebase, minimizing the diff.

Reply:
Makes sense 👍